server/test/unit/server/managers/auth/permission/test_cud.py

```diff
@@ -126,8 +126,38 @@ def test_non_existing_user_permission_revoke(self):
         try:
             self.permission_manager.revoke(r, login, [o])
             self.fail('Non-existing user permission revoke did not raise an exception')
-        except exceptions.MissingResource, e:
-            self.assertTrue(login in str(e))
+        except exceptions.InvalidValue, e:
+            self.assertTrue('login' in str(e))
+
+    def test_non_existing_user_permission_grant(self):
+        login = 'non-existing-user-login'
+        r = self._create_resource()
+        o = authorization.READ
+        try:
+            self.permission_manager.grant(r, login, [o])
+            self.fail('Non-existing user permission grant did not raise an exception')
+        except exceptions.InvalidValue, e:
+            self.assertTrue('login' in str(e))
+
+    def test_non_existing_role_permission_revoke(self):
```

Comment:
I figured these would be in a different test module. By convention this file would test `auth/permission/cud.py` but these tests testing code in `auth/role/cud.py`. Can these tests be moved to a module at `auth/role/test_cud.py`?